server/src/main/java/org/elasticsearch/monitor/metrics/NodeMetrics.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.monitor.metrics;
+
+import org.elasticsearch.action.admin.cluster.node.stats.NodeStats;
+import org.elasticsearch.action.admin.indices.stats.CommonStatsFlags;
+import org.elasticsearch.common.util.SingleObjectCache;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.node.NodeService;
+import org.elasticsearch.telemetry.metric.LongWithAttributes;
+import org.elasticsearch.telemetry.metric.MeterRegistry;
+
+import java.util.Map;
+import java.util.function.Supplier;
+
+public class NodeMetrics {
+    private final NodeService nodeService;
+    private final NodeStatsCache nodeStatsCache;
+
+    public NodeMetrics(MeterRegistry meterRegistry, NodeService nodeService) {
+        this.nodeService = nodeService;
+        this.nodeStatsCache = new NodeStatsCache(TimeValue.timeValueSeconds(1));
```

Comment:
For a minute I was concerned about this refreshing every 1 second, but after reading through the cache code I see that it lazy-loads. Nice 🙂

Reply:
Thanks! :D